subsys/bluetooth/controller/ll_sw/nordic/lll/lll_adv.c

```diff
@@ -99,6 +100,8 @@ static inline bool isr_rx_ci_adva_check(uint8_t tx_addr, uint8_t *addr,
 			     PAYLOAD_FRAG_COUNT))
 #define PDU_MEM_COUNT      (PDU_MEM_COUNT_MIN + PDU_MEM_FIFO_COUNT)
 #define PDU_POOL_SIZE      (PDU_MEM_SIZE * PDU_MEM_COUNT)
+/* TODO: make configurable or calculate sane value */
+#define LINK_POOL_SIZE     (16)
```

Comment:
@andrzej-kaczmarek  As acquire and release both are in the thread context,  memq data structure is not required for chaining. Use simple singly linked list to chain ADV PDU buffers. This will save RAM and reduce the code complexity.